File: solutions/_1845_seat_reservation_manager/solution.java

package _1845_seat_reservation_manager;

import java.util.PriorityQueue;

class SeatManager {
    PriorityQueue<Integer> availableSeats;

    public SeatManager(int n) {
        availableSeats = new PriorityQueue<>();
        for (int i = 1; i <= n; i++) {
            availableSeats.add(i);
        }
    }

    public int reserve() {
        return availableSeats.poll();
    }

    public void unreserve(int seatNumber) {
        availableSeats.add(seatNumber);
    }
}

/*
 * https://leetcode.com/problems/seat-reservation-manager/
 */

File: solutions/_1845_seat_reservation_manager/solution.js
const { MinPriorityQueue } = require("datastructures-js");

class SeatManager {
  constructor(n) {
    this.availableSeats = MinPriorityQueue.fromArray(
      Array.from({ length: n }, (_, i) => i + 1)
    );
  }

  reserve() {
    const seat = this.availableSeats.dequeue();
    return seat;
  }

  unreserve(seatNumber) {
    this.availableSeats.enqueue(seatNumber);
  }
}

module.exports = { SeatManager };

/*
https://leetcode.com/problems/seat-reservation-manager/
*/
